tests/test.cpp

```diff
@@ -2943,6 +2943,25 @@ void StructUnionTest() {
   gadget.Set(fan);
 }
 
+void WarningsAsErrorsTest() {
+  {
+    flatbuffers::IDLOptions opts;
+    opts.warnings_as_errors = false;
```

Comment:
nit: maybe avoid explicitly setting this so we can test that the default value is correct.

Reply:
Good idea!